# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1432336377 14400
#      Fri May 22 19:12:57 2015 -0400
# Node ID 4e15a4c331e73ccf9899875dd851db33d52003b3
# Parent  561af1ab60990a990c70c279475b621846639a83
doc: Fix griddata example to new calling convention

* geometry.txi (Interpolation on Scattered Data): Fix griddata example.
* geometryimages.m: Fix griddata call to match documented example and
produce the correct figure.

diff --git a/doc/interpreter/geometry.txi b/doc/interpreter/geometry.txi
--- a/doc/interpreter/geometry.txi
+++ b/doc/interpreter/geometry.txi
@@ -428,17 +428,18 @@ An example of the use of the @code{gridd
 
 @example
 @group
 rand ("state", 1);
 x = 2*rand (1000,1) - 1;
 y = 2*rand (size (x)) - 1;
 z = sin (2*(x.^2+y.^2));
 [xx,yy] = meshgrid (linspace (-1,1,32));
-griddata (x,y,z,xx,yy);
+zz = griddata (x, y, z, xx, yy);
+mesh (xx, yy, zz);
 @end group
 @end example
 
 @noindent
 that interpolates from a random scattering of points, to a uniform grid.
 @ifnotinfo
 The output of the above can be seen in @ref{fig:griddata}.
 
diff --git a/doc/interpreter/geometryimages.m b/doc/interpreter/geometryimages.m
--- a/doc/interpreter/geometryimages.m
+++ b/doc/interpreter/geometryimages.m
@@ -60,17 +60,18 @@ function geometryimages (nm, typ)
     triplot (tri, x, y);
     print ([nm "." typ], d_typ);
   elseif (strcmp (nm, "griddata"))
     rand ("state", 1);
     x = 2 * rand (1000,1) - 1;
     y = 2 * rand (size (x)) - 1;
     z = sin (2 * (x.^2 + y.^2));
     [xx,yy] = meshgrid (linspace (-1,1,32));
-    griddata (x,y,z,xx,yy);
+    zz = griddata (x, y, z, xx, yy);
+    mesh (xx, yy, zz);
     print ([nm "." typ], d_typ);
   elseif (strcmp (nm, "convhull"))
     x = -3:0.05:3;
     y = abs (sin (x));
     k = convhull (x, y);
     plot (x(k),y(k),'r-', x,y,'b+');
     axis ([-3.05, 3.05, -0.05, 1.05]);
     print ([nm "." typ], d_typ);
